source/common/router/rds_impl.cc

```diff
@@ -174,5 +184,44 @@ Router::RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getRouteCon
   return new_provider;
 };
 
+void RouteConfigProviderManagerImpl::addRouteInfo(RouteConfigProviderSharedPtr provider,
+                                                  Buffer::Instance& response) {
+  response.add(fmt::format(
+      "route_config_name: {}\n",
+      std::static_pointer_cast<RdsRouteConfigProviderImpl>(provider)->route_config_name_));
```

Comment:
Per @htuch you shouldn't be using static_cast here (and note even if you did you should be using dynamic_cast). Refactor interfaces as needed to make it work properly.